mypy/semanal.py

```diff
@@ -393,9 +393,10 @@ def refresh_top_level(self, file_node: MypyFile) -> None:
         self.export_map[self.cur_mod_id] = self.all_exports
         self.all_exports = []
 
-    def add_implicit_module_attrs(self, file_node: MypyFile) -> None:
-        """Manually add implicit definitions of module '__name__' etc."""
-        for name, t in implicit_module_attrs.items():
+    def add_implicit_attrs(self, file_node: MypyFile) -> None:
+        """Manually add implicit definitions of module and class '__name__' etc."""
+        implicit_attrs = {**implicit_module_attrs, **implicit_class_attrs}
+        for name, t in implicit_attrs.items():
```

Comment:
Note that `__qualname__` is not defined at module scope. It should only be added in class bodies. Also, it's only defined in Python 3.

Reply:
Ok, I will work on that.